### ISSUE TITLE
Ran into some issues installing in Linux apps environment on a Chromebook

### ISSUE THREAD

**Issue by espoelstra** [CircuitMess/CircuitBlocks#5]
I tried both the .deb package and the AppImage and I'm getting the same error on both. It appears to be when the application tries to request elevated access to install the Arduino components. It could be handy to document how to install those separately to work around this issue.

![Screenshot 2019-11-04 at 8 08 13 PM](https://user-images.githubusercontent.com/7798975/68174535-1ebbe480-ff44-11e9-9e93-b8eee8559dfb.png)
![Screenshot 2019-11-04 at 7 45 41 PM](https://user-images.githubusercontent.com/7798975/68174536-1ebbe480-ff44-11e9-8e5f-3daa3777734d.png)

**Comment by filipbudisa** [comment]
Hey there!

We don't have a Chromebook available for testing and honestly didn't event think anyone would try to run CircuitBlocks on Chome OS. Kudos to you for actually getting it to run! From the images attached, it seems you don't have any authentication agents installed. On linux systems, an auth agent is needed to display a super user prompt so CircuitBlocks can install the Arduino IDE and add required permissions so it can communicate with the phone. By default, on most linux distros, an auth agent comes prepackaged with the OS, but since you're running Chrome OS, you will need to do that manually. I'm afraid since non of us here at CircuitMess have any experience with Chromebook hacking, you're on your own here.
You can read more about about auth agents at https://wiki.archlinux.org/index.php/Polkit#Authentication_agents

As a workaround, you can manually install arduino by downloading it at https://www.arduino.cc/en/Main/Software and placing the arduino-x.x.x folder inside your ~/.arduino/ directory. That's where CB installs arduino and looks for it when launched. After that, run the arduino-linux-setup.sh as root and install.sh as a regular user. Those scripts can be found inside the downloaded arduino-x.x.x directory and they create a .desktop entry for the Arduino IDE and ensure proper permissions are set. Again, since you're doing this on Chrome OS, which isn't a regular linux distro, you may encounter some issues.

Happy hacking!

**Comment by dragon788** [comment]
I tried a couple authentication agents but didn't have much luck, so I had ended up trying exactly what you outlined, since the Arduino IDE had downloaded correctly and was sitting in the home directory.

I ended up getting the custom Ringo stuff imported but the arbitrary USB pass through into the Linux VM isn't quite fully working yet.

I've been looking at some of the other ways to get it working, and it sounds like WebUSB or WebSerial should be compatible with the ESP32 and would also enable programming from any device that runs Chrome, whether mobile devices or Windows or Linux or macOS.

**Comment by espoelstra** [comment]
I tried a couple authentication agents but didn't have much luck, so I had ended up trying exactly what you outlined, since the Arduino IDE had downloaded correctly and was sitting in the home directory.

I ended up getting the custom Ringo stuff imported but the arbitrary USB pass through into the Linux VM isn't quite fully working yet.

I've been looking at some of the other ways to get it working, and it sounds like WebUSB or WebSerial should be compatible with the ESP32 and would also enable programming from any device that runs Chrome, whether mobile devices or Windows or Linux or macOS.
